apis/v1beta1/config.go

```diff
@@ -387,6 +387,26 @@ func (c *Config) GetReadinessProbe(logger logr.Logger) (*corev1.Probe, error) {
 	return nil, nil
 }
 
+// GetStartupProbe gets the first enabled startup probe. There should only ever be one extension enabled
+// that provides the hinting for the startup probe.
+func (c *Config) GetStartupProbe(logger logr.Logger) (*corev1.Probe, error) {
+	if c.Extensions == nil {
+		return nil, nil
+	}
+
+	enabledComponents := c.GetEnabledComponents()
+	for componentName := range enabledComponents[KindExtension] {
+		// TODO: Clean up the naming here and make it simpler to use a retriever.
```

Comment:
Its still WIP?

Reply:
I'm not sure, this is copied over from `GetReadinessProbe` and `GetLivenessProbe`

Reply:
@frzifus this is a relic from my shameful methodology here...